src/validators/typed_dict.rs

```diff
@@ -163,10 +163,10 @@ impl Validator for TypedDictValidator {
         slots: &'data [CombinedValidator],
         recursion_guard: &'s mut RecursionGuard,
     ) -> ValResult<'data, PyObject> {
-        if let Some(field) = extra.assignee_field {
-            // we're validating assignment, completely different logic
-            return self.validate_assignment(py, field, input, extra, slots, recursion_guard);
-        }
+        let fields: Vec<(bool, &TypedDictField)> = match extra.updated_field {
+            Some((assignee_field_name, _)) => self.fields.iter().map(|f| (f.name != assignee_field_name, f)).collect(),
+            None => self.fields.iter().map(|f| (false, f)).collect(),
+        };
```

Comment:
I'm not a _huge_ fan of this pattern, but it is similar to what we had before while being less jarring in that it doesn't do "if a field on extra is set, call completely different logic". Ultimately it's a compromise between sharing code paths / reducing duplication between `validate` and `validate_assignment` and not injecting polymorphic behavior that depends on booleans.